src/components/Select.js

```diff
@@ -1,20 +1,32 @@
-// Elements must be a list of {value:, lable:} dictionaries
+// Elements must be a list of {value:, label:} dictionaries
 // e.g. const learnedLanguages = [
 //     { value: 'en', label: 'English' },
 //     { value: 'fr', label: 'French' },
 //     { value: 'da', label: 'Danish' }
 //   ]
 
-export default function Select({ elements, label, val, updateFunction }) {
+export default function Select({
+  elements,
+  label,
+  val,
+  updateFunction,
+  current,
+}) {
   return (
     <select onChange={(e) => updateFunction(e.target.value)}>
       <option style={{ display: "none" }} />
 
-      {elements.map((each) => (
-        <option key={val(each)} value={val(each)}>
-          {label(each)}
-        </option>
-      ))}
+      {elements.map((each) =>
```

Comment:
can we simplify this a bit more? ;)

Reply:
@mircealungu  Done! - Or? (See latest commit)

Reply:
nice :)